packages/editor/src/dataviews/actions/delete-permanently.ts

```diff
@@ -0,0 +1,120 @@
+/**
+ * WordPress dependencies
+ */
+import { store as coreStore } from '@wordpress/core-data';
+import { store as noticesStore } from '@wordpress/notices';
+import { __, sprintf } from '@wordpress/i18n';
+import type { Action } from '@wordpress/dataviews';
+
+/**
+ * Internal dependencies
+ */
+import { getItemTitle } from './utils';
+import type { Post } from '../types';
+import {
+	TEMPLATE_POST_TYPE,
+	TEMPLATE_PART_POST_TYPE,
+} from '../../store/constants';
+
+type CoreDataError = {
+	reason: { message?: string };
+};
+
+const permanentlyDeletePostAction: Action< Post > = {
```

Comment:
Hi @youknowriad, in order to know if the user has the right permissions to delete or trash a post, we need to call some selectors. Currently, we have regressions and are showing trash action even when we should not. I'm fixing that for Trash action at https://github.com/WordPress/gutenberg/pull/62589/ (which we should include for 6.6). We also need to do the same to delete permanently but we could not use a selector to check permissions in the API I guess we need to fix that at the API level.

Reply:
Also, something where we would check permissions to delete before registering the action would not work, the user may be able to delete some items (e.g.: the items they created or draft items) but not other items.

Reply:
The "callback" can receive the "registry" object, we might want to do the same for the isElligible.

Reply:
Hi @youknowriad, I think simply passing the registry to isElligible is not a solution and will not work as expected. In the beginning, while checks are being done canUser will return false, and if isEligible is false then canUser may say true but isEligible is not re-executed because there is nothing that causes a React rerender. This is not a theoretical issue it was happening on https://github.com/WordPress/gutenberg/pull/62589 and that was the reason I used the hacky solution of getCachedResolvers around useMemo to trigger the rerender.


I see two possible ways forward:
- Support hook-based actions on the API.
- Make is eligible something async and then it is up to the component users of isEligible to make sure the component rerender after isEligible finishes the computation.

Reply:
What if we just call isEligible within useSelect and pass "select". I'm not sure if I like it but it's a solution as well.

Reply:
> What if we just call isEligible within useSelect and pass "select". I'm not sure if I like it but it's a solution as well.

It is also a possible solution, it is going to be called a lot e.g: in a list with 50 times we will have 50 useSelects calling is eligible running on each store change. That may be ok. Another downside is that it is not very flexible it forces people to use WordPress data e.g.: I can not check if something is eligible on my own API without using resolvers etc.